js_modules/dagster-ui/packages/ui-core/src/runs/useRunsForTimeline.tsx

```diff
@@ -359,20 +392,45 @@ export const useRunsForTimeline = (
     );
 
     return jobs.sort((a, b) => earliest[a.key]! - earliest[b.key]!);
-  }, [workspaceOrError, runsByJobKey, start, end]);
+  }, [workspaceOrError, runsByJobKey, start, _end]);
+
+  const lastFetchRef = useRef({ongoing: 0, future: 0});
+  const lastRangeMs = useRef([0, 0] as readonly [number, number]);
+  if (Math.abs(lastRangeMs.current[0] - rangeMs[0]) > 30000) {
+    lastFetchRef.current = {ongoing: 0, future: 0};
+  }
+  lastRangeMs.current = rangeMs;
 
   const refreshState = useRefreshAtInterval({
     refresh: useCallback(async () => {
       await Promise.all([
-        fetchFutureTicks(),
+        // Only fetch ongoing runs once every 30 seconds
+        (async () => {
+          if (lastFetchRef.current.ongoing < Date.now() - 30 * 1000) {
+            await fetchOngoingRunsQueryData();
+            lastFetchRef.current.ongoing = Date.now();
+          }
+        })(),
+        // Only fetch future ticks on a minute
+        (async () => {
+          if (lastFetchRef.current.future < Date.now() - 60 * 1000) {
```

Comment:
nit: parens for clear precedence, or turn these in to consts

Reply:
I added parens but our eslint config removes them :(